File: Array&Hashing/containDuplicate/problem.py
def dup(list):

# __________________(brute force)________________________________
#     lenght=len(list)    
#     for i in range(lenght):
      
#         for j in range(lenght):
         
           
#          if(list[i]==list[j] and i!=j):
#                 return True
            
    
            
#     return False
           
            
# display=dup([1,2,2])

# print(display)
# __________________(optimize version)________________________________
    
    lenght=len(list)
    convertedset = set(list)
    set_len=len(convertedset)
    
    if(lenght==set_len):
        return False
    else:
        return True


display=dup([1,2,3,4,4])
print(display)



    
      
  

File: Array&Hashing/containDuplicate/problem.js
function hasDuplicate(nums) {
  // ____________(brute force)____________________

  //   for(let i= 0; i<=nums.length -1; i++){
  //     console.log("value i",i,nums[i]);

  //     for(let j=0; j<=nums.length-1; j++){
  //         console.log("value j",j, nums[j]);

  //         if(nums[i] == nums[j] && i != j){
  //         console.log(`i:${i}  j: ${j} and i value: ${nums[i]} j value: ${nums[j]}`);

  //             return true
  //         }

  //     }
  //   }

  //___________(optimize version)__________________

  let convertedset = [...new Set(nums)];

  let setlength = convertedset.length;
  console.log(setlength);
  let arrlength = nums.length;
  console.log(arrlength);

  if (setlength === arrlength) {
    return false;
  }

  return false;
}
console.log(hasDuplicate([1, 2, 2]));
